packages/core/_configurable.ts

```diff
@@ -39,3 +39,12 @@ export const defaultWindow = /* #__PURE__ */ isClient ? window : undefined
 export const defaultDocument = /* #__PURE__ */ isClient ? window.document : undefined
 export const defaultNavigator = /* #__PURE__ */ isClient ? window.navigator : undefined
 export const defaultLocation = /* #__PURE__ */ isClient ? window.location : undefined
+
+export interface ConfigurableDeepRefs<D extends boolean> {
+  /**
+   * Return deep refs instead of shallow refs.
+   *
+   * @default true - will be changed to `false` by default in the next major
+   */
+  deep?: D
```

Comment:
Oh I remember why I picked `shallow` before, as `deep` actually conflicts with watch's deep. Maybe name it explicitly as `deepRef`?

Reply:
Actually this should be fine, no? If we have a `deepRef` and we watch it, we might want to end up with `deep: true` anyway.

For most composables, we do not really extend `WatchOptions` (except for watch composables). And even if we do, in most cases we want to do it in a controlled way.

Reply:
Yeah, while it might make sense to do both, I am worried that having one flag controlling two things could result in footgun in future design. Consider what we have for `immediate`. I think it would better to keep on the safe path and avoid making premature assumptions of the usage cases.

> I think if we are going to expose watchOptions strait, we should do options.watchOptions?

that would be a bit too late to change that, as most of composable already designed it in a flat way